app/jobs/regular/pull_hotlinked_images.rb

```diff
@@ -181,9 +181,14 @@ def should_download_image?(src, post = nil)
         # media was enabled, then we definitely want to redownload again otherwise
         # we end up reusing existing uploads which may be linked to many posts
         # already.
-        upload = Upload.consider_for_reuse(Upload.get_from_url(src), post)
+        upload = Upload.get_from_url(src)
+        return true if upload.blank?
 
-        return !upload.present?
+        if SiteSetting.secure_media && upload.access_control_post_id.present?
+          return true if upload.copied_from_other_post?(post) || upload.uploaded_before_secure_media_enabled?
```

Comment:
So... this code will copy secure images from one post to another, simply using the URL? 

Could someone use this to steal images which they're not supposed to have access to?

Reply:
@davidtaylorhq I will check this out, this is a good point. Definitely want to redownload if `uploaded_before_secure_media_enabled?` but yeah maybe not in the other case. Thanks for pointing this out!